pyqtgraph/graphicsItems/LegendItem.py

```diff
@@ -73,16 +75,39 @@ def __init__(self, size=None, offset=None, horSpacing=25, verSpacing=0,
         if size is not None:
             self.setGeometry(QtCore.QRectF(0, 0, self.size[0], self.size[1]))
 
+        if sampleType is not None:
+            assert issubclass(sampleType, GraphicsWidget)
```

Comment:
`assert`'s in functional code are generally not a great idea as they are bypassed if someone runs `python -O script.py

```bash
$ python --help

...
-O     : remove assert and __debug__-dependent statements; add .opt-1 before
         .pyc extension; also PYTHONOPTIMIZE=x
...
```

If we want to have this check (which I have no problem with), we should have an `if not issubclass...` followed by a `RuntimeError`

I'm totally open to other checks/errors to be raised, this is just what I thought of first.